html/user/openssl_sign_action.php

```diff
@@ -0,0 +1,134 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+
+// Needs to be defined, unknown use.
+check_get_args(array("tnow", "ttok"));
+
+// Check the user is online
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+// For some reason, I'm only able to use numbers, no letters.
+// Need to provide security, but full functionality is likewise desirable.
+$user_data = sanitize_tags(post_str("user_data", true));
+
+// Following line a potential bug.
+// $user_data = BoincDb::escape_string($user_data);
+
+//Has the user entered data with length > 0 & <= 256?
+if (($user_data) && (strlen($user_data) <= 256)) {
```

Comment:
Why exactly 256 limit?

Reply:
Initially a 256 limit to prevent passing 1 mil characters in to inflate the text download, likewise hash160's are less than 256.

We could increase this to a few thousand if we want to allow say PGP user data, however I don't imagine most external systems would use more than 256 characters.

What would you propose as an appropriate max character limit?

Reply:
Idk 1000? 4k?
Output of sha512 as hex is 128 chars, so current 256 limit is probably ok.